### PR TITLE
corecfg: fix proxy.* writing and add integration test

### PR DESCRIPTION
**Pull request by mvo5**
Based on https://github.com/snapcore/snapd/pull/3753

This adds the missing integration test for `snap set core proxy.{http,https,ftp}=foo`

### CODE REVIEW

File: corecfg/utils_test.go [hunk 1]
```diff
@@ -31,24 +31,35 @@ type utilsSuite struct{}
...
 
+func (s *utilsSuite) TestUpdateKeyValueStreamConfigNotInAllConfig(c *C) {
+	in := bytes.NewBufferString("")
+	newConfig := map[string]string{"unsupported-options": "cannot be set"}
+	supportedConfigKeys := map[string]bool{
+		"foo": true,
+	}
+
+	_, err := corecfg.UpdateKeyValueStream(in, supportedConfigKeys, newConfig)
+	c.Check(err, ErrorMatches, `cannot set unsupported configuration value \"unsupported-options"`)
```
Comment (zyga) [inline corecfg/utils_test.go]: The `\` character there looks odd, why is it there?

File: interfaces/builtin/core_support.go [hunk 2]
```diff
@@ -93,6 +93,12 @@ const coreSupportConnectedPlugAppArmor = `
 owner /boot/uboot/           r,
 owner /boot/uboot/config.txt rwk,
 owner /boot/uboot/config.txt.* rwk,
+
+# Allow read/write /etc/environment so that proxy configuration can
```
Comment (zyga) [inline interfaces/builtin/core_support.go]: Nice catch, does this mean the feature doesn't work in 2.27.2?

File: corecfg/utils_test.go (hunk 1 above)
Comment (mvo5) [inline corecfg/utils_test.go]: Ups, silly me, removed.

File: interfaces/builtin/core_support.go (hunk 2 above)
Comment (mvo5) [inline interfaces/builtin/core_support.go]: We don't have corecfg yet in 2.27.X

**Pull request merged**